tutorials/scripting/gdextension/gdextension_cpp_example.rst

```diff
@@ -0,0 +1,648 @@
+
+.. note::
+
+    You may need to add ``bits=64`` to the command on Windows or Linux.
+
+Creating a simple plugin
+------------------------
+
+Now it's time to build an actual plugin. We'll start by creating an empty Godot
+project in which we'll place a few files.
+
+Open Godot and create a new project. For this example, we will place it in a
+folder called ``demo`` inside our GDExtension module's folder structure.
+
+In our demo project, we'll create a scene containing a Node called "Main" and
+we'll save it as ``main.tscn``. We'll come back to that later.
+
+Back in the top-level GDExtension module folder, we're also going to create a
+subfolder called ``src`` in which we'll place our source files.
+
+You should now have ``demo``, ``godot-cpp``, and ``src``
+directories in your GDExtension module.
+
+Your folder structure should now look like this:
+
+.. code-block:: md
+    gdextension_cpp_example/
+    |
+    +--demo/                  # game example/demo to test the extension
+    |
+    +--godot-cpp/             # C++ bindings
+    |
+    +--src/                   # source code of the extension we are building
+
+In the ``src`` folder, we'll start with creating our header file for the
+GDExtension node we'll be creating. We will name it ``gdexample.h``:
+
+.. code-block:: C++
+
+    #ifndef GDEXAMPLE_H
+    #define GDEXAMPLE_H
+
+    #include <godot_cpp/classes/sprite2d.hpp>
+
+    namespace godot {
+
+    class GDExample : public Sprite2D {
+        GDCLASS(GDExample, Sprite2D)
+
+    private:
+        float time_passed;
+
+    protected:
+        static void _bind_methods();
+
+    public:
+        GDExample();
+        ~GDExample();
+
+        void _init(); // our initializer called by Godot
+
+        void _process(float delta);
+    };
+
+    }
+
+    #endif
+
+There are a few things of note to the above. We include ``sprite2d.hpp`` which
+contains bindings to the Sprite2D class. We'll be extending this class in our
+module.
+
+We're using the namespace ``godot``, since everything in GDExtension is defined
+within this namespace.
+
+Then we have our class definition, which inherits from our Sprite2D through a
+container class. We'll see a few side effects of this later on. The
+``GDCLASS`` macro sets up a few internal things for us.
+
+After that, we declare a single member variable called ``time_passed``.
+
+In the next block we're defining our methods, we obviously have our constructor
+and destructor defined, but there are two other functions that will likely look
+familiar to some, and one new method.
+
+The first is ``_bind_methods``, which is a static function that Godot will
+call to find out which methods can be called and which properties it exposes. 
+The second is our ``_process`` function, which will work exactly the same 
+as the ``_process`` function you're used to in GDScript. The third is 
+our ``_init`` function which is called after Godot has properly set up
+our object.
+
```

Comment:
What is the difference beween extensions  ``_init``  and gdscript  ``_init``

Reply:
I am not sure to be quite honest. There was this dicussion comment here: https://github.com/godotengine/godot-cpp/issues/348#issuecomment-742084113 saying:
> _init is null because in the Godot API, _init is a """""virtual""""" function. Not exactly in the C++ sense though: this is a function that exists in the API because Godot expects us to be able to implement it optionally, but it doesn't actually define a default implementation itself. The function does not even exists in the engine (or sometimes it does under a different definition), it's purely a script API artifact. I have yet to find a class of the API with a virtual method you can actually call.

Reply:
I would think all `Object`s are able to implement `_init` where the constructor is called when the cpp is created and the _init is called in the godot lifecycle. I would even know how to test the difference.
Let's keep it as is to remove confusion.

Reply:
I'm not 100% sure anymore, it's been awhile, @Faless you may be able to confirm.

In GDNative `_init` had to be defined because the wrapper code always called it. I don't think this is so anymore in GDExternal, I think it's optional.
It should now work the same as in GDScript where you define `_init` and register it in `bind_methods` and it will get called if needed.